### PR TITLE
Add uniqueName decorator

### PR DESCRIPTION
**Pull request by xaviervia**
Moving the uniqueName decorator from the [klarna/ui](https://github.com/klarna/ui/pull/147)

- [x] Add README docs
- [x] Add dependency
- [x] Add test

### CODE REVIEW

File: src/uniqueName.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(TOKENS, (c) => {
+    const r = random() * 16 | 0
+    const v = c === 'x' ? r : (r & 0x3 | 0x8)
+    return v.toString(16)
+  })
+}
+
+let counter = 0
+export default (Target) => {
+  class UniqueName extends PureComponent {
```
Comment (Deschtex) [inline src/uniqueName.js]: Shouldn't it return the class?

Comment (Deschtex) [inline src/uniqueName.js]: Shouldn't it return the class?

Comment (Nevon) [inline src/uniqueName.js]: That seems like a good idea.

File: src/uniqueName.spec.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import {render} from 'react-dom'
+import {equal, ok} from 'assert'
+import uniqueName from './uniqueName'
+
+describe('uniqueName', () => {
+  describe('no name specified', () => {
+    it('has unique id', () => {
```
Comment (Nevon) [inline src/uniqueName.spec.js]: You're not testing that it has a unique id, just that it has a name that may or may not be unique. Render a bunch of them and check that there are no collisions.

**Pull request merged**